### PR TITLE
Add Three Bull Shoals Datasets from Issue 320

### PR DESCRIPTION
**Pull request by padilla410**
This PR is a partial solution to #320 ([this](https://github.com/USGS-R/lake-temperature-model-prep/issues/320#issuecomment-1110015652) table) and adds data from the following cooperator files:
* 20190409 DATA with all depths.csv 
* Bull Shoals and LOZ profile data LMVP
* Temp_DO BSL * 

The data in these files comes from two locations: Bull Shoals Reservoir and Lake of the Ozarks. The Bull Shoals data was successfully added to the pipeline, while the Lake of the Ozarks data was not, due to the unresolved issue #289.

### A `case_when` question
I have a specific question for @jread-usgs. While working on these parsers I was trying to find spots to use regular expressions and `case_when`. Specifically, I tried six ways from Sunday to integrate [this](https://github.com/padilla410/lake-temperature-model-prep/blob/906e031ea730ce3136db12e37b84e91b228db3e0/7a_temp_coop_munge/src/data_parsers/parse_missouri_files.R#L97-L100) snippet into [this](https://github.com/padilla410/lake-temperature-model-prep/blob/906e031ea730ce3136db12e37b84e91b228db3e0/7a_temp_coop_munge/src/data_parsers/parse_missouri_files.R#L89-L94) portion of a `dplyr` pipeline.

Specifically, I thought this should work:
```r
# read in data and clean
clean <- list_sheets %>%
  purrr::map_df(~ readxl::read_xlsx(path = file_path, sheet = .x)) %>%
  purrr::discard(~all(is.na(.x))) %>%  # remove columns with all NA
  dplyr::filter(!is.na(Timestamp)) %>% # remove rows without timestamp
  dplyr::rename_with(~ new_col_names) %>%
  mutate(
    
    DateTime = case_when(
      class(DateTime) == 'character' ~ mdy_hm(DateTime) %>% as.Date,
      all(class(DateTime)) != 'character' ~ as.Date(DateTime),
    ),
    
    Timezone = c('CDT/CST'),
    temp = fahrenheit_to_celsius(temp),
    depth = -1 * depth,
    Missouri_ID = 'Missouri_100' # value from Univ MO xwalk (https://drive.google.com/file/d/11w6-LXCDSDCjipFYPxUgJyf7YB9BtXYR/view?usp=sharing)
  ) %>%
  dplyr::select(DateTime, Timezone, depth, temp, site, Missouri_ID)
```
The reason I need to have conditional formatting for `DateTime` is because the `DateTime` field reads as either `character` or `POSIXct` for each of the 14 files in `6_temp_coop_fetch/in/Temp_DO_BSL_MM_DD_YYYY.zip`.

### Results from scmake("8_viz")
Bull Shoals **BEFORE**
image

Bull Shoals **AFTER**
image

Summary numbers (no change compared to #333):
image

### CODE REVIEW

**Comment by lindsayplatt** [comment]
@padilla410 I am reviewing this now. Looking at your question about the use of `case_when()`, based on your prompt for it, it sounds like what you have does not work and you expected it to. Do you have any other info about what isn't working? That would help me troubleshoot a working solution.

File: 7a_temp_coop_munge/src/data_parsers/parse_missouri_files.R [hunk 1]
```diff
@@ -1,6 +1,113 @@
...
+parse_20190409_DATA_with_all_depths <- function(inind, outind) {
 
+  infile <- sc_retrieve(inind, remake_file = '6_temp_coop_fetch_tasks.yml')
+  outfile <- as_data_file(outind)
+
+  dat_raw <- readr::read_csv(infile)
+
+  # split date and time
+  dat_datetime_split <- stringr::str_split_fixed(dat_raw$Timestamp, " ", 2) %>%
+    as_tibble %>%
```
Comment (lindsayplatt) [inline 7a_temp_coop_munge/src/data_parsers/parse_missouri_files.R:11]: Do you not need `as_tibble()` here? Interesting!

File: 7a_temp_coop_munge/src/data_parsers/parse_missouri_files.R [hunk 2]
```diff
@@ -1,6 +1,113 @@
...
+read_bullshoals_data <- function(file_path, new_col_names, keep_cols) {
+
+  # identify sheets with data
+  list_sheets <- readxl::excel_sheets(file_path) %>%
+    .[grepl('^Pt', .)]
+
+  # read in data and clean
+  clean <- list_sheets %>%
+    purrr::map_df(~ readxl::read_xlsx(path = file_path, sheet = .x)) %>%
+    purrr::discard(~all(is.na(.x))) %>%  # remove columns with all NA
```
Comment (lindsayplatt) [inline 7a_temp_coop_munge/src/data_parsers/parse_missouri_files.R:86]: WOW!! I didn't know about `purrr::discard()` or `purrr::keep()`. Can't wait to use those in the future!

**Review by lindsayplatt — APPROVED** [review]
Looks good. I know you were trying to solve your DateTime fix using tidyverse and it seems like `case_when()` did not have the outcome you expected. Not sure what was going on but here is a working example that I think matches the use case you have?

```r
x <- tibble(datetime = c("2022-01-01 00:00:00"))
y <- tibble(datetime = as.POSIXct(c("2022-01-01 00:00:00")))

do_conversion <- function(df) {
  df %>% mutate(datetime_converted = case_when(
    is.character(datetime) ~ "CONVERTED",
    !is.character(datetime) ~ "LEFT ALONE"
  ))
}


> x %>% do_conversion()
# A tibble: 1 x 2
  datetime            datetime_converted
  <chr>               <chr>             
1 2022-01-01 00:00:00 CONVERTED         
> y %>% do_conversion()
# A tibble: 1 x 2
  datetime            datetime_converted
  <dttm>              <chr>             
1 2022-01-01 00:00:00 LEFT ALONE     
```

File: 7a_temp_coop_munge/src/data_parsers/parse_missouri_files.R (hunk 2 above)
Comment (padilla410) [reply 7a_temp_coop_munge/src/data_parsers/parse_missouri_files.R:86]: Glad to hear! I'm working on replacing my dependence on the `apply` family with `purrr` so I'm definitely digging into the doc.

**Pull request merged**